libr/core/cmd_debug.c

```diff
@@ -3405,13 +3405,26 @@ static void r_core_cmd_bp(RCore *core, const char *input) {
 			int sl = r_str_word_set0 (str);
 			addr = r_num_math (core->num, DB_ARG(0));
 			if (watch) {
-					if (sl == 2) {
-						rw = (strcmp (DB_ARG(1), "r") == 0 ? R_BP_PROT_READ : R_BP_PROT_WRITE);
-					} else {
-						eprintf ("Usage: dbw <addr> <rw> # Add watchpoint\n");
+				if (sl == 2) {
+					if (strcmp (DB_ARG(1), "r") == 0){
```

Comment:
Use ! Instead of ==0

Reply:
But i would reuse the r_str_rwx function to also handle x and -

Reply:
will change to ! instead of ==0.

Thanks for pinpointing to rwx.
Do you mean I should use the r_str_rwx in the compare somehow to cover watch and breaks at the same address? Or do you want to get rid of rw and use rwx based on the r_str_rwx to pass to r_debug_bp_add and then to parse the rwx within r_debug_bp? Would make it more common from my point of view.

Reply:
the more generic the better imho. parsing rwx is pretty generic, so i would just parse that expression with r_str_rwx and then check the bits if they are set or not